service/tablet_allocator.cc

```diff
@@ -346,8 +346,10 @@ class load_balancer {
     using global_shard_id = tablet_replica;
     using shard_id = seastar::shard_id;
 
-    // Represents metric for per-node load which we want to equalize between nodes.
-    // It's an average per-shard load in terms of tablet count.
+    // Represents metric for load which we want to equalize between shards or nodes.
+    // Load balancer equalizes storage utilization.
+    // It is assumed that each tablet has equal size and that shards and nodes can have different capacity.
+    // So we equalize: tablet_count * target_tablet_size / capacity_in_bytes.
```

Comment:
We can consider a metric that mixes absolute free space and relative free space, not just relative free space as here.

Relative free space makes sense when considering that a shard with twice the capacity will see twice the ingest rate.

Absolute free space makes sense when considering non-proportional overhead such as group0 work.

But out of scope for now, relative free space is fine for i4i, maybe not for things like m7gd.

Reply:
Why do you think group0 work (which work?) is relevant for balancing?